### PR TITLE
Offboard type mask fix

### PR DESCRIPTION
**Pull request by Seanmatthews**

### CODE REVIEW

**Comment by Seanmatthews** [comment]
Follow up from #7401

**Comment by mhkabir** [comment]
You seem to have old commits in this branch. Is this intentional?

**Comment by dagar** [comment]
FYI there was interest in creating enums. https://github.com/mavlink/mavlink/pull/670

**Comment by Seanmatthews** [comment]
@mhkabir Yes.

@dagar Yup, someone expressed interest in it, but it's a larger project that will touch more parts of the system. The intention of this PR is only to fix an obvious bug in the system.

**Comment by TSC21** [comment]
@Seanmatthews can you squash and rebase this? Also, is it flight tested?

**Comment by TSC21** [comment]
Jenkins test this please

**Comment by TSC21** [comment]
@Seanmatthews can you shows an example of this working? On SITL perhaps?

**Comment by TSC21** [comment]
@Seanmatthews we are still waiting that you fix up the commits into a single one

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -861,7 +861,7 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
 
 		bool is_takeoff_sp = (bool)(set_position_target_local_ned.type_mask & 0x1000);
 		bool is_land_sp = (bool)(set_position_target_local_ned.type_mask & 0x2000);
-		bool is_loiter_sp = (bool)(set_position_target_local_ned.type_mask & 0x3000);
+		bool is_loiter_sp = (set_position_target_local_ned.type_mask & 0x3000) == 0x3000;
```
Comment (AlexisTM) [inline src/modules/mavlink/mavlink_receiver.cpp]: Why would it need this change?

**Comment by julianoes** [comment]
The crucial commit https://github.com/PX4/Firmware/pull/7693/commits/897ec276be9b45dbe82c76ce048c19f08905895a has actually been merged to master a while ago.

**Pull request closed without merging**